packages/core/src/Stores/client-store.js

```diff
@@ -1140,8 +1146,26 @@ export default class ClientStore extends BaseStore {
     }
 
     @action.bound
-    setDeviceData(device_data) {
-        this.device_data = device_data;
+    setDeviceData() {
+        // Set client URL params on init
+        const url_params = new URLSearchParams(window.location.search);
+        const device_data = {
+            ...(url_params.get('affiliate_token') && { affiliate_token: url_params.get('affiliate_token') }),
+            ...(this.device_data.date_first_contact && {
```

Comment:
Shouldn't this be `!this.device_data.date_first_contact`? As we don't want to overwrite the first-contact date if we already have it, right?